app/src/main/java/com/example/cleanarchitecturesample/util/dispatchers/DispatcherProvider.kt

```diff
@@ -0,0 +1,10 @@
+package com.example.cleanarchitecturesample.util.dispatchers
+
+import kotlinx.coroutines.CoroutineDispatcher
+
+interface DispatcherProvider {
+    val main: CoroutineDispatcher
+    val io: CoroutineDispatcher
+    val default: CoroutineDispatcher
+    val unconfined: CoroutineDispatcher
+}
```

Comment:
Is there a need to provide all dispatchers? From what I understand, we just need to provide a mechanism where we could have different dispatchers for `production` and `testing` and we're already able to achieve it by using dependency injection i.e. inject `Default or IO` for prod and `Main` for testing. Do we still need such an interface?

Reply:
> Majority of the code LGTM, let's conclude the Dispatcher comment and we should be good to go 👍

thanks, doing the changes